### PR TITLE
HTML-824: enrollinprogram tag needs to expand method for determining …

### PR DESCRIPTION
**Pull request by mogoodrich**
…encounter date

## Issue I worked on

see https://issues.openmrs.org/browse/HTML-

## Checklist: I completed these to help reviewers :)

- [ ] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/htmlformentry/FormEntryContext.java [hunk 1]
```diff
@@ -930,29 +930,26 @@ public void registerPropertyAccessorInfo(String property, String widgetId, Strin
 	 */
 	public Date getBestApproximationOfEncounterDate() {
 		// if there's a pending update to the encounter date, use that
-		Date encounterDateToTest = getPendingEncounterDatetime();
+		Date encounterDatetime = getPendingEncounterDatetime();
```
Comment (mogoodrich) [inline api/src/main/java/org/openmrs/module/htmlformentry/FormEntryContext.java:933]: tangential change... the variable name was annoying me...

Comment (mseaton) [reply api/src/main/java/org/openmrs/module/htmlformentry/FormEntryContext.java:933]: Changing the variable name is fine, I agree

File: api/src/main/java/org/openmrs/module/htmlformentry/FormEntryContext.java [hunk 2]
```diff
@@ -930,29 +930,26 @@ public void registerPropertyAccessorInfo(String property, String widgetId, Strin
...
 			// If there is a visit in the context and the current date is outside of this visit, use visit start date
-			if (getVisit() != null) {
-				Visit visit = (Visit) getVisit();
-				if (visit.getStopDatetime() != null && currentDate.after(visit.getStopDatetime())) {
-					encounterDateToTest = visit.getStartDatetime();
-				}
+			if (getVisit() != null && ((Visit) getVisit()).getStopDatetime() != null
+			        && currentDate.after(((Visit) getVisit()).getStopDatetime())) {
+				encounterDatetime = ((Visit) visit).getStartDatetime();
 			} else {
```
Comment (mseaton) [inline api/src/main/java/org/openmrs/module/htmlformentry/FormEntryContext.java:946]: So, I actually can't see any meaningful difference in this code?  What exactly did you change?  Stylistically, I don't think this is an improvement, I prefer the code that was in place before.  What am I missing?

Comment (mogoodrich) [reply api/src/main/java/org/openmrs/module/htmlformentry/FormEntryContext.java:946]: The issue is when line 944 evaluates to true and 946 evaluates to false, line 951 never gets hit, and null gets returned.   Alternatively to my change, we could add another else at 948 that also sets encounterDate = currentDate.

Or we could rework the whole thing just to return at each point when it finds a match, but I didn't to have to retest all use cases after doing that.

Comment (mseaton) [reply api/src/main/java/org/openmrs/module/htmlformentry/FormEntryContext.java:946]: Oh, I see.  Yeah, I would have just removed the else block and just had a new "if encounter date == null, set to xxx" at the end.  But this is fine.

**Review by mseaton — APPROVED** [review]

**Pull request merged**